src/ray/core_worker/actor_manager.h

```diff
@@ -196,6 +200,11 @@ class ActorManager {
   /// getting them from GCS frequently.
   absl::flat_hash_map<std::string, ActorID> cached_actor_name_to_ids_
       GUARDED_BY(cache_mutex_);
+
+  /// Set of subscribed actors.
+  absl::flat_hash_set<ActorID> subscribed_actors_;
```

Comment:
Why not just using `mutex_ `?

Reply:
> Why not just using `mutex_ `?

The `subscription_mutex_` is only used to protect `subscribed_actors_`, and make sure the subscription is only invoked once, it's a fine grained lock. It will bring some cost if use `mutex_`, because each actor task will visit `subscribed_actors_`, the `mutex_` coarse grained in a way.